### PR TITLE
[RV64_DYNAREC] Fixed some opcodes caught by cosim

### PR DESCRIPTION
**Pull request by xctan**

### CODE REVIEW

File: src/dynarec/rv64/dynarec_rv64_emit_shift.c [hunk 1]
```diff
@@ -1429,6 +1435,9 @@ void emit_shld32(dynarec_rv64_t* dyn, int ninst, rex_t rex, int s1, int s2, int
     } else IFX(X_ALL) {
         SET_DFNONE();
     }
+
+    BEQZ_MARK(s5);
```
Comment (ksco) [inline src/dynarec/rv64/dynarec_rv64_emit_shift.c:1439]: It seems for now `emit_shld32` only gets called once in `SHLD Ed, Gd, CL`, and there is already a `BEQZ_NEXT` on s5 before the function call.

**Comment by ptitSeb** [comment]
So it's ready to merge?

File: src/dynarec/rv64/dynarec_rv64_00_2.c [hunk 2]
```diff
@@ -547,7 +547,7 @@ uintptr_t dynarec64_00_2(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int
         case 0x9F:
             INST_NAME("LAHF");
             READFLAGS(X_CF | X_PF | X_AF | X_ZF | X_SF);
-            ANDI(x1, xFlags, 0xFF);
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_00_2.c:550]: well, that OF2 flags doesn't seems handled in this opcode?

Comment (xctan) [reply src/dynarec/rv64/dynarec_rv64_00_2.c:550]: OF2 has taken the place of res3 but it is not read here, so the mask below will strip the extra OF2, as I confirmed in the cosim.

**Comment by xctan** [comment]
> So it's ready to merge?

Yup. I cannot find any other prominent differences between dynarec and the interpreter when running VMP with cosim.

File: src/dynarec/rv64/dynarec_rv64_00_2.c (hunk 2 above)
Comment (ptitSeb) [reply src/dynarec/rv64/dynarec_rv64_00_2.c:550]: ah yes, OF is to high to get there. All good then :)

**Pull request merged**